presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConnectorFactory.java

```diff
@@ -51,6 +56,21 @@ public ConnectorHandleResolver getHandleResolver()
     @Override
     public Connector create(String catalogName, Map<String, String> config, ConnectorContext context)
     {
-        return createConnector(catalogName, config, context, Optional.empty());
+        ClassLoader classLoader = context.duplicatePluginClassLoader();
+        try {
+            Class<?> hiveConnectorFactoryClass = classLoader.loadClass(HIVE_CONNECTOR_FACTORY);
+            checkState(hiveConnectorFactoryClass != null, "Could not load %s from duplicate plugin class loader", HIVE_CONNECTOR_FACTORY);
+
+            Method method = hiveConnectorFactoryClass.getMethod("createConnector", String.class, Map.class, ConnectorContext.class, Optional.class);
+            return (Connector) method.invoke(null, catalogName, config, context, Optional.empty());
```

Comment:
set `classLoader` as TCCL

Reply:
This isn't necessary.  The `InternalHiveConnectorFactory.createConnector` method handles class loaders correctly.

Reply:
Still it's much safer to have thi

Reply:
Why is it safer? It’s part of the contract of the method. If the class loaders were different, then setting it here wouldn’t do anything as the method overwrites it.